homeassistant/components/amcrest/__init__.py

```diff
@@ -197,14 +197,18 @@ def _wrap_test_online(self, now):
 
 
 def _monitor_events(hass, name, api, event_codes):
-    event_codes = ",".join(event_codes)
+    event_codes = set(event_codes)
     while True:
         api.available_flag.wait()
         try:
-            for code, start in api.event_actions(event_codes, retries=5):
-                signal = service_signal(SERVICE_EVENT, name, code)
-                _LOGGER.debug("Sending signal: '%s': %s", signal, start)
-                dispatcher_send(hass, signal, start)
+            for code, start in api.event_actions("All", retries=5):
+                event_data = {"camera": name, "event": code, "payload": start}
+                _LOGGER.debug("Firing amcrest event: %s", event_data)
```

Comment:
The core will log this too at debug level so we don't need it here.

Reply:
Fixed